src/ultk/language/grammar.py

```diff
@@ -558,7 +561,12 @@ def from_module(cls, module_name: str) -> "Grammar":
         """
         module = import_module(module_name)
         grammar = cls(None)
-        for name, value in inspect.getmembers(module):
+        possible_rules = ()
```

Comment:
Minor things: (i) you shouldn't need to define this separately here, since it will be defined in both the `if` and the `else` block below.  (ii) Unlike lists and dictionaries, `tuple()` is necessary to define an empty tuple, not merely parentheses

Reply:
Ah ok, variable scoping in python messes me up occasionally. I have removed this line. Thank you.